ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -246,6 +250,30 @@ public TransientQueryMetadata executeQuery(
     }
   }
 
+  @Override
+  public PullQueryResult executePullQuery(
```

Comment:
Another question for my own education: what's the difference between this class and `QueryEngine`? It seems the latter is only for persistent and transient push queries?